File: Utils/Forms/리트코드 양식.js
// https://leetcode.com/problems/count-number-of-nice-subarrays/

/*
constraints:
*/


/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var numberOfSubarrays = function(nums, k) {

};


const inputDatas = [
    {data: [], answer: ""},
];

/*
LeetCode Hard.
제출 230.2K, 정답률 51.6%
*/

for (let i = 0; i < inputDatas.length; i++) {
    const {data, answer} = inputDatas[i];
    const res = numberOfSubarrays(...data);
    if (JSON.stringify(res) === JSON.stringify(answer)) {
        console.log("pass");
    } else {
        let summary = "fail";
        for (const [label, content] of [["expected:", answer], ["got:", res]]) {
            summary += `\n  ${label}\n`;
            summary += `    ${content}`;
            summary = summary.trimEnd();
        }
        console.log(summary);
    }
}

File: Utils/Forms/리트코드 양식.rs
// https://leetcode.com/problems/exclusive-time-of-functions/
struct Solution;

/*
constraints:
*/

impl Solution {
    pub fn exclusive_time(n: i32, logs: Vec<String>) -> Vec<i32> {
        return vec![];
    }
}

#[macro_export]
macro_rules! run_judge {
    (
        $func:path,
        [ $( { data: ( $($arg:expr),* ), answer: $ans:expr } ),* $(,)? ]
    ) => {
        $(
            let result = $func($($arg),*);
            let expected = $ans;
            if result == expected {
                println!("pass");
            } else {
                let mut summary = String::from("fail");
                for (label, content) in [("expected:", &expected), ("got:", &result)] {
                    summary.push_str(&format!("
  {}
", label));
                    summary.push_str(&format!("    {:?}", content));
                }
                summary = summary.trim_end().to_string();
                println!("{}", summary);
            }
        )*
    };
}

#[allow(unused_macros)]
macro_rules! svec {
    ($($x:expr),* $(,)?) => {
        vec![$($x.to_string()),*]
    };
}

fn main() {
    run_judge!(
        Solution::exclusive_time,
        [
            {data: (,), answer: vec![]},
            {data: (,), answer: vec![]},
            {data: (,), answer: vec![]}
        ]
    );
}

/*
LeetCode Medium.
제출 507.6K, 정답률 66.1%
*/
